qipr_approver/approver/views/shib.py

```diff
@@ -0,0 +1,23 @@
+from django.shortcuts import render
+from django.http import HttpResponse, Http404
+from django.views.decorators.csrf import csrf_protect
+
+from approver.workflows import shib
+from approver.constants import SHIB_ENABLED
+
+#TODO: implement real shib and get rid of this fake shib endpoint
+@csrf_protect
+def shib_login(request):
+    if SHIB_ENABLED == 'false':
+        if request.method == "GET":
+            return render(request, 'approver/shib.html')
+        elif request.method == "POST":
+            return shib.after_validation(request)
+        else:
+            raise Http404('This url only supports POST and GET.')
+
+    else:
+        if (request.META.get('HTTP_EPPN') and SHIB_ENABLED == 'true'):
+            return shib.after_validation(request)
+        else:
+            raise Http404('There is a problem with your Shibboleth django settings')
```

Comment:
Id swap these blocks around a bit to make it clearer. 

Putting the `if SHIB_ENABLED` first makes it easier to grok because less negatives and its what will be happening in prod. Also you wouldnt need to do the double check of SHIB_ENABLED on line 20.